### ISSUE TITLE
Feature request: Allow module attributes

### ISSUE THREAD

**Issue by kleinesfilmroellchen** [lalrpop/lalrpop#686]
My use case for this is to turn off various lints for the LALRPOP generated file, as it violates a lot of pedantic/nursery clippy hints that I like to have in the rest of my project.

Currently, using a module attribute at the beginning of a file:

```rust
#![allow(clippy::all, clippy::pedantic, clippy::nursery)]

use some_module::ImportantStruct;
// ...

grammar;

// ...
```

causes this error:

```
error: an inner attribute is not permitted in this context
 --> out/generated.rs:3:1
  |
3 | #![allow(clippy::all, clippy::pedantic, clippy::nursery)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4 | use some_module::ImportantStruct;
  | --------------------------------- the inner attribute doesn't annotate this `use` import
  |
  = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files
help: to annotate the `use` import, change the attribute from inner to outer style
  |
3 - #![allow(clippy::all, clippy::pedantic, clippy::nursery)]
3 + #[allow(clippy::all, clippy::pedantic, clippy::nursery)]
  |
```

This is confusing, because the generated file correctly contains the module attribute at the beginning of the file, just below the `sha3` line. However, the generated code contains its module attributes (which includes an `allow(clippy::all)`) at the beginning of the contained `mod __parse__GlobalNonterminal`, where `GlobalNonTerminal` is the (only) `pub`-declared non-terminal of the grammar. Therefore, it should probably be possible to move these attributes down in in the generated code. After all, it seems like LALRPOP parses them correctly at the beginning of files; in other places they cause a syntax error.

**Comment by Pat-Lafon** [comment]
> Is this still reproducible? I've done something similar with https://github.com/sampsyo/bril/blob/main/bril-rs/bril2json/src/bril_grammar.lalrpop and afaik that works as expected(I've also been trying to get lalrpop to pass more of these lints). How do you include this file so I can try it out?

To answer my own question. I didn't have this issue because I was using the `generate_in_source_tree` configuration option which allows for this. If you use `lalrpop_mod` then this is a problem.

**Comment by kleinesfilmroellchen** [comment]
This is not fixed, I get the exact same error.

**Comment by Pat-Lafon** [comment]
Yeah, to clarify, I think this is actually an issue with the `include!` macro. These issues seem to be related: https://github.com/rust-lang/rust/issues/66920 https://github.com/rust-lang/rust/issues/117464

**Comment by lujobi** [comment]
While not as flexible as your suggestion: Passing the `allow(clippy::all)` attributes to the `lalrpop_mod!` macro as done in #463 might be suitable option.

**Comment by kleinesfilmroellchen** [comment]
> While not as flexible as your suggestion: Passing the `allow(clippy::all)` attributes to the `lalrpop_mod!` macro as done in [#463](https://github.com/lalrpop/lalrpop/issues/463) might be suitable option - at least as long as you do not use the `generate_in_source_tree` option.

That is what I am doing right now, so it’s an acceptable solution for the time being.
